added info about limitations to set script (#18756)
demisto/content: Packs/CommonScripts/Scripts/Set/README.md

## Human Readable Output

>Key Data(val.ID == obj.ID) set

## Known Limitations
The script has some limitations with appending to the context in different cases and formats:

> - When *append* is set to `false` - the script will not support [dt](https://xsoar.pan.dev/docs/integrations/dt)operations given as part of the *key* argument.

Reviewer: ```suggestion
- When *append* is set to `false` - the script will not support [dt](https://xsoar.pan.dev/docs/integrations/dt) operations given as part of the *key* argument.
```

- When **Set** is called from within another script - if *append* is set tot `true` it is necessary to manually
return the results given by the `demisto.executeCommand`, for the context update to take place.
For example:
    ```python

>     res = demisto.executeCommand('Set', {'append': 'true', 'key': 'MyKey', 'value': 'MyValue'})

Reviewer: Sorry I forgot we encourage `execute_command` on top of `demisto.executeCommand` (it does the error handling for you) 
```suggestion
    res = execute_command('Set', {'append': 'true', 'key': 'MyKey', 'value': 'MyValue'})
```

## Human Readable Output

>Key Data(val.ID == obj.ID) set

> ## Known Limitations
> The script has some limitations with appending to the context in different cases and formats:
> If *append* is set to false - the script support append in a regular format (and not dt).
> If *append* is set to true - the script doesn't automatically updates the context in a case called from another script, but only after returning its result by adding `return_results` to the command `demisto.executeCommand`.
> For example: ```return_results(demisto.executeCommand('Set', <args>))```.

Reviewer: ```suggestion
## Known Limitations
The script has some limitations with appending to the context in different cases and formats:
- When *append* is set to `false` - the script will not support [dt](https://xsoar.pan.dev/docs/integrations/dt) operations given as part of the *key* argument.
- When **Set** is called from within another script - if *append* is set tot `true` it is necessary to manually return the results given by the `demisto.executeCommand`, for the context update to take place.
For example: 
    ```python
    res = demisto.executeCommand('Set', {'append': 'true', 'key': 'MyKey', 'value': 'MyValue'})
    return_results(res)
    ```
```
